BaseButton: make input related information in the tooltip optional

lupoDharkael opened the pull request:
solves https://github.com/godotengine/godot/issues/23617

akien-mga commented:
Won't that remove all shortcut hints from the editor buttons?

lupoDharkael commented:
I didn't think about it. I don't think a tooltip should be imposed but the lack of automatic editor tooltips about the shortcuts is inconvenient.
What about having conditional show of the shortcuts in the tooltip?

CowThing commented:
Yeah that's true, I forgot about that, the editor does use this for shortcut hints in buttons. I think the real problem is that InputEventAction's get_text() function returns a string that isn't any help for shortcut hints, because it doesn't actually show which buttons trigger the action, and has a lot of extra text on it. InputEventKey's get_text() function just returns the key and modifiers, with no extra text, making it much better for hints.

An option to disable the shortcut tooltip would be nice if there's no easy way to show which inputs trigger a specific InputEventAction.

lupoDharkael committed:
BaseButton: add shortcut_in_tooltip

This flag disables the input related information in the tooltip.
It is exposed as a member variable in gdscript.

lupoDharkael commented:
I've reimplemented this PR, now it exposes a variable in the BaseButton class to disable the input related info in the tooltip.
Tell me if you agree with the name of the methods and the variable exposed to gdscript so I can start adding this to the docs in this PR.

akien-mga commented on scene/gui/base_button.cpp:
```diff
@@ -535,6 +547,7 @@ void BaseButton::_bind_methods() {
 	ADD_SIGNAL(MethodInfo("toggled", PropertyInfo(Variant::BOOL, "button_pressed")));
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "disabled"), "set_disabled", "is_disabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "toggle_mode"), "set_toggle_mode", "is_toggle_mode");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "show_input_tooltip"), "set_show_input_tooltip", "is_show_input_tooltip_enabled");
```
Maybe `tooltip_with_shortcut_info`?

akien-mga replied:
Or `add_shortcut_to_tooltip`? It's hard to find something not too long but still descriptive enough :D

akien-mga replied:
`shortcut_in_tooltip` maybe.

lupoDharkael replied:
`shortcut_in_tooltip` seems like a good option.

lupoDharkael commented:
I've added it to the docs.

akien-mga commented:
Thanks!
